test/regression/generateRegressionTests.ts

```diff
@@ -6,10 +6,10 @@ import {test, TestContext} from "ava";
 
```

Comment:
Make a story to split this whole file up into several files. Separate by browser. `generate{Chrome,Firefox,ETC}RegressionTests.ts`.

Reply:
https://jira.rax.io/browse/SURF-1085

Reply:
This also needs to be split up in the following use cases:

1. All of the components for a single browser (or a subset of browsers)
2. All of the browsers for a single component (or a subset of components)

```
yarn test:regression --browsers=chrome,firefox --components=buttons,typography/test/typefaces.html
```

If you pass in an html file, go ahead and run just that one test. If you pass in something else, it's a directory and any html file inside that directory will be run for the given browser profile.

You need to be able to re-run just a single failing test. Currently, there's only one way to do that, and that's because we use one big snap file for everything. The architecture needs to default to single tests, and have a top-level test run simply enumerate through each individual test, instead of generating the full suite every time.

Finally, double check Ava's docs about running concurrent tests and make sure we're maxing out our saucelabs quota on every run.

Finally, more docs.